[Stats] Use summary stats endpoint to fetch total visitor count

hichamboushaba opened the pull request:
Closes: #11171 

### Description
This PR uses https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2983 to use the summary stats endpoint to fetch the visitor's total count; the count is fetched only for predefined ranges.

In this PR, I also made changes to the granularity we use for `2 days` ranges, please check p1711724047631339-slack-C03L1NF1EA3 for more information.

### Testing instructions
I don't have any store that has enough visitor stats to showcase the difference between the `views` and `visitors`, but I think that's OK for now.

1. Open the app.
2. Test the predefined ranges, and confirm total visitors is shown.
3. Optionally: compare the visitor stats count with what's shown in Web and iOS.
4. Create a custom range with a single day.
5. Confirm the visitor's count is shown.
6. Create a custom range with 2 or more days.
7. Confirm the visitor's count is hidden.
8. Tap on a graph point.
9. Confirm the visitor's count is shown.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->


- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Bump fluxc

hichamboushaba committed:
Add repository functions to fetch total visitor stats

hichamboushaba committed:
Update logic to fetch total visitors count

hichamboushaba committed:
Fix granularity and date of summary stats

hichamboushaba committed:
Show visitors total count from the summary stats

hichamboushaba committed:
Use cached summary stats only when not force refreshing

hichamboushaba committed:
Use correct flag for force refreshing

hichamboushaba committed:
Make function that returns Flow not suspendable

hichamboushaba committed:
Fix and add unit tests

dangermattic commented:
<!--
  0 Errors
  3 Warnings: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/204">18.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba committed:
Adjust condition of when to hide visitor count for custom ranges

We were assuming that less than 2 days is equivalent to a single day, but that's not the case, as the end date is always picked at the end of the day and is inclusive.

hichamboushaba committed:
Adjust stats granularity to use "days" for 2 days' ranges.

hichamboushaba committed:
Add mock files of the summary stats endpoint

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/analytics/ranges/StatsTimeRangeGranularities.kt:
```diff
@@ -35,3 +35,22 @@ val StatsTimeRangeSelection.visitorStatsGranularity: StatsGranularity
+        SelectionType.LAST_MONTH, SelectionType.MONTH_TO_DATE -> StatsGranularity.MONTHS
+        SelectionType.LAST_YEAR, SelectionType.YEAR_TO_DATE -> StatsGranularity.YEARS
+        SelectionType.LAST_QUARTER, SelectionType.QUARTER_TO_DATE ->
+            error("Summary visitor stats unsupported quarter ranges")
```
@atorresveiga this PR adds support for using the summary stats endpoint for visitors count, for now only in the MyStore screen.
When you try to integrate this to the Analytics Hub, please take a note of this limitation, the endpoint doesn't support the `quarter` periods, so you'll have to use something like what iOS are [using](https://github.com/woocommerce/woocommerce-ios/blob/a919ffc19eb6e0f3240707a1cf77a222d604d198/Yosemite/Yosemite/Stores/StatsStoreV4.swift#L250-L268), which is basically "summing up the monthly durations", this won't be as accurate as the monthly one, but it'll be close, see pe5uwI-5c-p2

JorgeMucientes commented:
Great job, @hichamboushaba. The code looks clean and works well. 
The only issue I noticed is that we are not properly handling the errors when fetching stats. If the users doesn't have access to store stats like `shop manager` roles, then the visitor stats won't even show the dash. As mentioned over Slack, fixing this is out of scope for this PR so we can probably create another GH issue for it. 


https://github.com/woocommerce/woocommerce-android/assets/2663464/55f72973-2e6e-414e-8aa1-afb0265629bf

hichamboushaba committed:
Bump fluxc

hichamboushaba committed:
Merge branch 'trunk' into issue/11171-visitor-summary-stats

# Conflicts:
#	build.gradle

hichamboushaba committed:
Add release notes entry
